src/terminal/adapter/charsets.hpp

```diff
@@ -19,7 +19,7 @@ namespace Microsoft::Console::VirtualTerminal
     public:
         constexpr CharSet(const std::initializer_list<std::pair<wchar_t, wchar_t>> replacements)
         {
-            for (auto i = L'\0'; i < _translationTable.size(); i++)
+            for (size_t i = L'\0'; i < _translationTable.size(); i++)

```

Comment:
FYI, I don't think this change compiles. It might make more sense to cast the table size if that'll satisfy the CodeQL scan, i.e. something like this:
```cpp
for (auto i = L'\0'; i < gsl::narrow_cast<wchar_t>(_translationTable.size()); i++)
```
There's no risk of an infinite loop here anyway because this code is executed at compile time.